src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -352,95 +352,94 @@ public bool PostProcess(PublishRoot root)
 
         private bool PrunePackages(PublishRoot root)
         {
-            var resolver = new DefaultPackagePathResolver(root.TargetPackagesPath);
-
-            // Special cases (for backwards compat)
-            var specialFolders = new List<string> {
-                "native",
-                "InteropAssemblies",
-                "redist",
-                "runtimes"
-            };
-
-            if (!root.NoSource)
+            if (root.MainProjectLockFile == null)
             {
-                // 'shared' folder is build time dependency, so we only copy it when deploying with source
-                specialFolders.Add("shared");
+                return true;
             }
 
+            var resolver = new DefaultPackagePathResolver(root.TargetPackagesPath);
+
             var lockFilePath = Path.GetFullPath(Path.Combine(ApplicationBasePath, LockFileFormat.LockFileName));
-            var format = new LockFileFormat();
-            root.LockFile = format.Read(lockFilePath);
+            root.PublishedLockFile = new LockFileFormat().Read(lockFilePath);
 
-            var keep = new HashSet<string>();
+            var filesToKeep = new HashSet<string>();
+            var filesToRemove = new HashSet<string>();
 
-            foreach (var target in root.LockFile.Targets)
+            foreach (var target in root.PublishedLockFile.Targets)
             {
                 foreach (var library in target.Libraries)
                 {
-                    var packagesDir = resolver.GetInstallPath(library.Name, library.Version);
-                    var manifest = resolver.GetManifestFilePath(library.Name, library.Version);
-
-                    keep.Add(manifest);
+                    var packageDir = resolver.GetInstallPath(library.Name, library.Version);
+                    
+                    if (library.Name != root.MainProjectName && library.Type == "package")
+                    {
+                        filesToRemove.Add(resolver.GetHashPath(library.Name, library.Version));
```

Comment:
I'm assuming all package dependencies have a hash and package file. If this is not a fair assumption I'll check for existence of these files before adding them to `filesToRemove`